src/Sentry.Unity.iOS/SentryNativeCocoa.cs

```diff
@@ -58,6 +59,21 @@ internal static void Configure(SentryUnityOptions options, ISentryUnityInfo sent
             if (sentryUnityInfo.IL2CPP)
             {
                 options.DefaultUserId = SentryCocoaBridgeProxy.GetInstallationId();
+                if (string.IsNullOrEmpty(options.DefaultUserId))
+                {
+                    // In case we can't get an installation ID we create one and sync that down to the native layer
+                    options.DiagnosticLogger?.LogDebug("Failed to fetch 'Installation ID' from the native SDK. Creating new 'Default User ID'.");
+
+                    options.DefaultUserId = AnalyticsSessionInfo.userId;
```

Comment:
`AnalyticsSessionInfo.userId`

Is this something devs would need to add to their app manifest now? Might be a problem to use something that can be used to track users across apps. That's why we use our own `guid` everywhere

Reply:
I picked `AnalyticsSessionInfo.userId` as we currently use it for the desktop platforms. It's a GUID generated during installation. Based on the [Unity Docs] it only gets regenerated during a re-install. (https://docs.unity3d.com/2018.3/Documentation/ScriptReference/Analytics.AnalyticsSessionInfo-userId.html)

For the Unity SDK to use the .NET SDKs installation ID we'd need to make that public.
https://github.com/getsentry/sentry-dotnet/blob/d88635187ba02b4e47906a38b5bc0ebb19548355/src/Sentry/SentryOptions.cs#L46

Reply:
oh OK, so it uses the same principal as we do.

Worth adding a big block of comment clarifying that btw for anyone reasing the code and being worried about tracking and privacy